vim_menutrans_helper.py

```diff
@@ -0,0 +1,196 @@
+                    mid_part = line[pre_end + 1: ite.start() - 1].strip()
+                    if not ((len(mid_part) == 1 and mid_part[0] == ".") or len(mid_part) == 0 or pre_end == -1):
+                        temp_word_list.append("5")
+                    pre_end = ite.end()
+                    temp_word_list.append(line[ite.start() + 1: ite.end() - 1])
+
+                temp_line = ""
+                for word in temp_word_list:
+
+                    if len(word) == 0:
+                        continue
+                    if word[0] == "'":
+                        word = unescape_single_quotes(word)
+                    else:
+                        word = unescape_double_quotes(word)
+                    temp_line += word
+                line = temp_line
+
+            word_list = []
+            # split line string according to blank without backslash or tab ahead
+            for ite in re.finditer(r"(?:(?:\\.)|[^ \t])+", line):
+                word_list.append(line[ite.start(): ite.end()])
+
+            if len(word_list) == 0 or word_list[0] == '"':
+                continue
+            index = 0
+            # menu_item with "disable" or enable followed close behind should be ignored
+            if word_list[0] in menu_item and len(word_list) > 1 and (word_list[1] == "disable" or word_list[1] == "enable"):
+                continue
+
+            if word_list[0] == 'sil!':
+                index += 1
+            # blocks needs to be translated without being splited by '.'
+            if index < len(word_list) and word_list[index] in menu_item:
+                tobe_translated_blocks = ""
+                index += 1
+                while index < len(word_list):
+                    if word_list[index] == '<silent>' or word_list[index] == '<script>' or (word_list[index][0].isdigit() and not word_list[index - 1][0].isdigit()):
+                        index += 1
+                    else:
+                        # menu_item with "ToolBar" has no following words to be translated
+                        if not ("ToolBar." in word_list[index]):
+                            tobe_translated_blocks = word_list[index]
+                        break
+
+                # string starting with sill! needs to be removed first digit
+                if len(tobe_translated_blocks) > 0 and tobe_translated_blocks[0] == '5':
+                    tobe_translated_blocks = tobe_translated_blocks[1:]
+                if len(tobe_translated_blocks) > 0 and tobe_translated_blocks[-1] == '5':
+                    tobe_translated_blocks = tobe_translated_blocks[:-1]
+
+                # menu_item with "ToolBar" constructed a tobe_translated_blocks with length equals zero
+                if len(tobe_translated_blocks) == 0:
+                    continue
+
+                # untranslated word string splitted by "."
+                tobe_translated_word = tobe_translated_blocks.split('.')
```

Comment:
Splitting the string with dot all the way and then re-construct it back to handle the escaped dot is too hacky, and rather error prone. Could you please use regex to do the string splitting instead?